src/freenet/client/filter/ElementInfo.java

```diff
@@ -427,11 +427,16 @@ public static boolean isBannedPseudoClass(String cname)
 				// Pseudo-classes can be chained, at least dynamic ones can, see CSS2.1 section 5.11.3
 				String[] split = cname.split(":");
 				for(String s : split)
-					if(isBannedPseudoClass(s)) return true;
+					if(isBannedPseudoClass2(s)) return true;
```

Comment:
why the switch to isBannedPseudoClass2 ?

Reply:
isBannedPseudoClass2 is a newly added function, in order to avoid `String[] split = cname.split(":");` again for just a single class and improve performance.

Reply:
Nice! Thank you!